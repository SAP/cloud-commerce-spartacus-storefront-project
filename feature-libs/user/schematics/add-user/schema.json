{
  "$schema": "http://json-schema.org/schema",
  "id": "UserSchematics",
  "title": "User Schematics",
  "description": "Most of the Spartacus features require User Account feature to be properly configured.",
  "type": "object",
  "properties": {
    "project": {
      "type": "string",
      "description": "The name of the project.",
      "$default": {
        "$source": "projectName"
      }
    },
    "lazy": {
      "type": "boolean",
      "description": "Lazy load the user features.",
      "default": true
    },
    "features": {
      "type": "array",
      "uniqueItems": true,
      "items": {
        "enum": ["Account", "Profile"],
        "type": "string"
      },
      "default": ["Account", "Profile"],
      "x-prompt": "Which User features would you like to setup? Please note that most of the Spartacus features require User Account feature to be properly configured."
    }
  },
  "required": []
}
