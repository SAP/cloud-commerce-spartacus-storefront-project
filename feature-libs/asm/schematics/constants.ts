export const ASM_MODULE = 'AsmModule';
export const ASM_FEATURE_NAME = 'asm';
export const ASM_ROOT_MODULE = 'AsmRootModule';
export const SPARTACUS_ASM = '@spartacus/asm';
export const SPARTACUS_ASM_ROOT = `${SPARTACUS_ASM}/root`;
export const SPARTACUS_ASM_ASSETS = `${SPARTACUS_ASM}/assets`;
export const ASM_TRANSLATIONS = 'asmTranslations';
export const ASM_TRANSLATION_CHUNKS_CONFIG = 'asmTranslationChunksConfig';

export const CLI_ASM_FEATURE = 'asm';
