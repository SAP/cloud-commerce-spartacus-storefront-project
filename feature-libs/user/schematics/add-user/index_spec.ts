/// <reference types="jest" />

import {
  SchematicTestRunner,
  UnitTestTree,
} from '@angular-devkit/schematics/testing';
import {
  Schema as ApplicationOptions,
  Style,
} from '@schematics/angular/application/schema';
import { Schema as WorkspaceOptions } from '@schematics/angular/workspace/schema';
import {
  LibraryOptions as SpartacusPersonalizationOptions,
  SpartacusOptions,
} from '@spartacus/schematics';
import * as path from 'path';
import { CLI_ACCOUNT_FEATURE, CLI_PROFILE_FEATURE } from '../constants';

const collectionPath = path.join(__dirname, '../collection.json');
const userFeatureModulePath =
  'src/app/spartacus/features/user/user-feature.module.ts';
const scssFilePath = 'src/styles/spartacus/user.scss';

// TODO: Improve tests after lib-util test update
describe('Spartacus User schematics: ng-add', () => {
  const schematicRunner = new SchematicTestRunner('schematics', collectionPath);

  let appTree: UnitTestTree;

  const workspaceOptions: WorkspaceOptions = {
    name: 'workspace',
    version: '0.5.0',
  };

  const appOptions: ApplicationOptions = {
    name: 'schematics-test',
    inlineStyle: false,
    inlineTemplate: false,
    routing: false,
    style: Style.Scss,
    skipTests: false,
    projectRoot: '',
  };

  const spartacusDefaultOptions: SpartacusOptions = {
    project: 'schematics-test',
    configuration: 'b2c',
    lazy: true,
    features: [],
  };

  const defaultFeatureOptions: SpartacusPersonalizationOptions = {
    project: 'schematics-test',
    lazy: true,
    features: [CLI_ACCOUNT_FEATURE, CLI_PROFILE_FEATURE],
  };

  beforeEach(async () => {
    schematicRunner.registerCollection(
      '@spartacus/schematics',
      '../../projects/schematics/src/collection.json'
    );

    appTree = await schematicRunner
      .runExternalSchematicAsync(
        '@schematics/angular',
        'workspace',
        workspaceOptions
      )
      .toPromise();
    appTree = await schematicRunner
      .runExternalSchematicAsync(
        '@schematics/angular',
        'application',
        appOptions,
        appTree
      )
      .toPromise();
    appTree = await schematicRunner
      .runExternalSchematicAsync(
        '@spartacus/schematics',
        'ng-add',
        { ...spartacusDefaultOptions, name: 'schematics-test' },
        appTree
      )
      .toPromise();
  });

  describe('Without features', () => {
    beforeEach(async () => {
      appTree = await schematicRunner
        .runSchematicAsync(
          'ng-add',
          { ...defaultFeatureOptions, lazy: false, features: [] },
          appTree
        )
        .toPromise();
    });

    it('should not add any modules', () => {
      expect(appTree.exists(userFeatureModulePath)).toBeFalsy();
    });
  });

  describe('Account feature', () => {
    describe('styling', () => {
      beforeEach(async () => {
        appTree = await schematicRunner
          .runSchematicAsync('ng-add', defaultFeatureOptions, appTree)
          .toPromise();
      });

      it('should create a proper scss file', () => {
        const scssContent = appTree.readContent(scssFilePath);
        expect(scssContent).toMatchSnapshot();
      });

      it('should update angular.json', async () => {
        const content = appTree.readContent('/angular.json');
        expect(content).toMatchSnapshot();
      });
    });

    describe('eager loading', () => {
      beforeEach(async () => {
        appTree = await schematicRunner
          .runSchematicAsync(
            'ng-add',
            { ...defaultFeatureOptions, lazy: false },
            appTree
          )
          .toPromise();
      });

      it('should import appropriate modules', async () => {
        const module = appTree.readContent(userFeatureModulePath);
        expect(module).toContain(
          `import { UserAccountRootModule } from "@spartacus/user/account/root";`
        );
        expect(module).toContain(
          `import { UserAccountModule } from "@spartacus/user/account";`
        );
      });

      it('should not contain lazy loading syntax', async () => {
        const module = appTree.readContent(userFeatureModulePath);
        expect(module).not.toContain(`import('@spartacus/user/account').then(`);
      });
    });

    describe('lazy loading', () => {
      beforeEach(async () => {
        appTree = await schematicRunner
          .runSchematicAsync('ng-add', defaultFeatureOptions, appTree)
          .toPromise();
      });

      it('should import UserAccountRootModule and contain the lazy loading syntax', async () => {
        const module = appTree.readContent(userFeatureModulePath);
        expect(module).toContain(
          `import { UserAccountRootModule, USER_ACCOUNT_FEATURE } from "@spartacus/user/account/root";`
        );
        expect(module).toContain(`import('@spartacus/user/account').then(`);
      });

      it('should not contain the UserAccountModule import', () => {
        const module = appTree.readContent(userFeatureModulePath);
        expect(module).not.toContain(
          `import { UserAccountModule } from "@spartacus/user/account";`
        );
      });
    });
  });

  describe('Profile feature', () => {
    describe('eager loading', () => {
      beforeEach(async () => {
        appTree = await schematicRunner
          .runSchematicAsync(
            'ng-add',
            { ...defaultFeatureOptions, lazy: false },
            appTree
          )
          .toPromise();
      });

      it('should import appropriate modules', async () => {
        const module = appTree.readContent(userFeatureModulePath);
        expect(module).toContain(
          `import { UserProfileRootModule } from "@spartacus/user/profile/root";`
        );
        expect(module).toContain(
          `import { UserProfileModule } from "@spartacus/user/profile";`
        );
      });

      it('should not contain lazy loading syntax', async () => {
        const module = appTree.readContent(userFeatureModulePath);
        expect(module).not.toContain(`import('@spartacus/user/profile').then(`);
      });
    });

    describe('lazy loading', () => {
      beforeEach(async () => {
        appTree = await schematicRunner
          .runSchematicAsync('ng-add', defaultFeatureOptions, appTree)
          .toPromise();
      });

      it('should import UserProfileRootModule and contain the lazy loading syntax', async () => {
        const module = appTree.readContent(userFeatureModulePath);
        expect(module).toContain(
          `import { UserProfileRootModule, USER_PROFILE_FEATURE } from "@spartacus/user/profile/root";`
        );
        expect(module).toContain(`import('@spartacus/user/profile').then(`);
      });

      it('should not contain the UserAccountModule import', () => {
        const module = appTree.readContent(userFeatureModulePath);
        expect(module).not.toContain(
          `import { UserAccountModule } from "@spartacus/user/profile";`
        );
      });
    });
  });
});
