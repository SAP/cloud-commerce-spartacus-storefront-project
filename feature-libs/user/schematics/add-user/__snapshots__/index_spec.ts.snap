// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spartacus User schematics: ng-add Account feature eager loading should import appropriate modules 1`] = `
"import { NgModule } from '@angular/core';
import { I18nConfig, provideConfig } from \\"@spartacus/core\\";
import { UserAccountModule } from \\"@spartacus/user/account\\";
import { userAccountTranslationChunksConfig, userAccountTranslations } from \\"@spartacus/user/account/assets\\";
import { UserAccountRootModule } from \\"@spartacus/user/account/root\\";
import { UserProfileModule } from \\"@spartacus/user/profile\\";
import { userProfileTranslationChunksConfig, userProfileTranslations } from \\"@spartacus/user/profile/assets\\";
import { UserProfileRootModule } from \\"@spartacus/user/profile/root\\";

@NgModule({
  declarations: [],
  imports: [
    UserAccountRootModule,
    UserAccountModule,
    UserProfileRootModule,
    UserProfileModule
  ],
  providers: [provideConfig(<I18nConfig>{
    i18n: {
      resources: userAccountTranslations,
      chunks: userAccountTranslationChunksConfig,
    },
  }),
  provideConfig(<I18nConfig>{
    i18n: {
      resources: userProfileTranslations,
      chunks: userProfileTranslationChunksConfig,
    },
  })
  ]
})
export class UserFeatureModule { }
"
`;

exports[`Spartacus User schematics: ng-add Account feature general setup should add the feature using the lazy loading syntax 1`] = `
"import { NgModule } from '@angular/core';
import { CmsConfig, I18nConfig, provideConfig } from \\"@spartacus/core\\";
import { userAccountTranslationChunksConfig, userAccountTranslations } from \\"@spartacus/user/account/assets\\";
import { UserAccountRootModule, USER_ACCOUNT_FEATURE } from \\"@spartacus/user/account/root\\";
import { userProfileTranslationChunksConfig, userProfileTranslations } from \\"@spartacus/user/profile/assets\\";
import { UserProfileRootModule, USER_PROFILE_FEATURE } from \\"@spartacus/user/profile/root\\";

@NgModule({
  declarations: [],
  imports: [
    UserAccountRootModule,
    UserProfileRootModule
  ],
  providers: [provideConfig(<CmsConfig>{
    featureModules: {
      [USER_ACCOUNT_FEATURE]: {
        module: () =>
          import('@spartacus/user/account').then((m) => m.UserAccountModule),
      },
    }
  }),
  provideConfig(<I18nConfig>{
    i18n: {
      resources: userAccountTranslations,
      chunks: userAccountTranslationChunksConfig,
    },
  }),
  provideConfig(<CmsConfig>{
    featureModules: {
      [USER_PROFILE_FEATURE]: {
        module: () =>
          import('@spartacus/user/profile').then((m) => m.UserProfileModule),
      },
    }
  }),
  provideConfig(<I18nConfig>{
    i18n: {
      resources: userProfileTranslations,
      chunks: userProfileTranslationChunksConfig,
    },
  })
  ]
})
export class UserFeatureModule { }
"
`;

exports[`Spartacus User schematics: ng-add Account feature general setup should install necessary Spartacus libraries 1`] = `
"{
  \\"name\\": \\"workspace\\",
  \\"version\\": \\"0.0.0\\",
  \\"scripts\\": {
    \\"ng\\": \\"ng\\",
    \\"start\\": \\"ng serve\\",
    \\"build\\": \\"ng build\\",
    \\"test\\": \\"ng test\\",
    \\"lint\\": \\"ng lint\\",
    \\"e2e\\": \\"ng e2e\\"
  },
  \\"private\\": true,
  \\"dependencies\\": {
    \\"@angular/animations\\": \\"~10.2.4\\",
    \\"@angular/common\\": \\"~10.2.4\\",
    \\"@angular/compiler\\": \\"~10.2.4\\",
    \\"@angular/core\\": \\"~10.2.4\\",
    \\"@angular/forms\\": \\"~10.2.4\\",
    \\"@angular/platform-browser\\": \\"~10.2.4\\",
    \\"@angular/platform-browser-dynamic\\": \\"~10.2.4\\",
    \\"@angular/router\\": \\"~10.2.4\\",
    \\"@angular/service-worker\\": \\"^10.1.0\\",
    \\"@ng-bootstrap/ng-bootstrap\\": \\"^7.0.0\\",
    \\"@ng-select/ng-select\\": \\"^5.0.9\\",
    \\"@ngrx/effects\\": \\"^10.0.0\\",
    \\"@ngrx/router-store\\": \\"^10.0.0\\",
    \\"@ngrx/store\\": \\"^10.0.0\\",
    \\"@spartacus/assets\\": \\"^3.2.0-next.2\\",
    \\"@spartacus/core\\": \\"^3.2.0-next.2\\",
    \\"@spartacus/storefront\\": \\"^3.2.0-next.2\\",
    \\"@spartacus/styles\\": \\"^3.2.0-next.2\\",
    \\"@spartacus/user\\": \\"^3.2.0-next.2\\",
    \\"angular-oauth2-oidc\\": \\"^10.0.1\\",
    \\"bootstrap\\": \\"^4.0\\",
    \\"i18next\\": \\"^19.3.4\\",
    \\"i18next-xhr-backend\\": \\"^3.2.2\\",
    \\"ngx-infinite-scroll\\": \\"^8.0.0\\",
    \\"rxjs\\": \\"~6.6.0\\",
    \\"tslib\\": \\"^2.0.0\\",
    \\"zone.js\\": \\"~0.10.2\\"
  },
  \\"devDependencies\\": {
    \\"@angular-devkit/build-angular\\": \\"~0.1002.1\\",
    \\"@angular-devkit/schematics\\": \\"^10.2.1\\",
    \\"@angular/cli\\": \\"~0.5.0\\",
    \\"@angular/compiler-cli\\": \\"~10.2.4\\",
    \\"@types/node\\": \\"^12.11.1\\",
    \\"@types/jasmine\\": \\"~3.5.0\\",
    \\"@types/jasminewd2\\": \\"~2.0.3\\",
    \\"codelyzer\\": \\"^6.0.0\\",
    \\"jasmine-core\\": \\"~3.6.0\\",
    \\"jasmine-spec-reporter\\": \\"~5.0.0\\",
    \\"karma\\": \\"~5.0.0\\",
    \\"karma-chrome-launcher\\": \\"~3.1.0\\",
    \\"karma-coverage-istanbul-reporter\\": \\"~3.0.2\\",
    \\"karma-jasmine\\": \\"~4.0.0\\",
    \\"karma-jasmine-html-reporter\\": \\"^1.5.0\\",
    \\"protractor\\": \\"~7.0.0\\",
    \\"ts-node\\": \\"~8.3.0\\",
    \\"tslint\\": \\"~6.1.0\\",
    \\"typescript\\": \\"~4.0.2\\"
  }
}
"
`;

exports[`Spartacus User schematics: ng-add Account feature general setup styling should create a proper scss file 1`] = `"@import \\"@spartacus/user\\";"`;

exports[`Spartacus User schematics: ng-add Account feature general setup styling should update angular.json 1`] = `
"{
  \\"$schema\\": \\"./node_modules/@angular/cli/lib/config/schema.json\\",
  \\"version\\": 1,
  \\"newProjectRoot\\": \\"\\",
  \\"projects\\": {
    \\"schematics-test\\": {
      \\"projectType\\": \\"application\\",
      \\"schematics\\": {
        \\"@schematics/angular:component\\": {
          \\"style\\": \\"scss\\"
        }
      },
      \\"root\\": \\"\\",
      \\"sourceRoot\\": \\"src\\",
      \\"prefix\\": \\"app\\",
      \\"architect\\": {
        \\"build\\": {
          \\"builder\\": \\"@angular-devkit/build-angular:browser\\",
          \\"options\\": {
            \\"outputPath\\": \\"dist/schematics-test\\",
            \\"index\\": \\"src/index.html\\",
            \\"main\\": \\"src/main.ts\\",
            \\"polyfills\\": \\"src/polyfills.ts\\",
            \\"tsConfig\\": \\"tsconfig.app.json\\",
            \\"aot\\": true,
            \\"assets\\": [
              \\"src/favicon.ico\\",
              \\"src/assets\\"
            ],
            \\"styles\\": [
              \\"src/styles.scss\\",
              \\"src/styles/spartacus/user.scss\\"
            ],
            \\"scripts\\": []
          },
          \\"configurations\\": {
            \\"production\\": {
              \\"fileReplacements\\": [
                {
                  \\"replace\\": \\"src/environments/environment.ts\\",
                  \\"with\\": \\"src/environments/environment.prod.ts\\"
                }
              ],
              \\"optimization\\": true,
              \\"outputHashing\\": \\"all\\",
              \\"sourceMap\\": false,
              \\"extractCss\\": true,
              \\"namedChunks\\": false,
              \\"extractLicenses\\": true,
              \\"vendorChunk\\": false,
              \\"buildOptimizer\\": true,
              \\"budgets\\": [
                {
                  \\"type\\": \\"initial\\",
                  \\"maximumWarning\\": \\"2mb\\",
                  \\"maximumError\\": \\"5mb\\"
                },
                {
                  \\"type\\": \\"anyComponentStyle\\",
                  \\"maximumWarning\\": \\"6kb\\",
                  \\"maximumError\\": \\"10kb\\"
                }
              ]
            }
          }
        },
        \\"serve\\": {
          \\"builder\\": \\"@angular-devkit/build-angular:dev-server\\",
          \\"options\\": {
            \\"browserTarget\\": \\"schematics-test:build\\"
          },
          \\"configurations\\": {
            \\"production\\": {
              \\"browserTarget\\": \\"schematics-test:build:production\\"
            }
          }
        },
        \\"extract-i18n\\": {
          \\"builder\\": \\"@angular-devkit/build-angular:extract-i18n\\",
          \\"options\\": {
            \\"browserTarget\\": \\"schematics-test:build\\"
          }
        },
        \\"test\\": {
          \\"builder\\": \\"@angular-devkit/build-angular:karma\\",
          \\"options\\": {
            \\"main\\": \\"src/test.ts\\",
            \\"polyfills\\": \\"src/polyfills.ts\\",
            \\"tsConfig\\": \\"tsconfig.spec.json\\",
            \\"karmaConfig\\": \\"karma.conf.js\\",
            \\"assets\\": [
              \\"src/favicon.ico\\",
              \\"src/assets\\"
            ],
            \\"styles\\": [
              \\"src/styles.scss\\",
              \\"src/styles/spartacus/user.scss\\"
            ],
            \\"scripts\\": []
          }
        },
        \\"lint\\": {
          \\"builder\\": \\"@angular-devkit/build-angular:tslint\\",
          \\"options\\": {
            \\"tsConfig\\": [
              \\"tsconfig.app.json\\",
              \\"tsconfig.spec.json\\",
              \\"e2e/tsconfig.json\\"
            ],
            \\"exclude\\": [
              \\"**/node_modules/**\\"
            ]
          }
        },
        \\"e2e\\": {
          \\"builder\\": \\"@angular-devkit/build-angular:protractor\\",
          \\"options\\": {
            \\"protractorConfig\\": \\"e2e/protractor.conf.js\\",
            \\"devServerTarget\\": \\"schematics-test:serve\\"
          },
          \\"configurations\\": {
            \\"production\\": {
              \\"devServerTarget\\": \\"schematics-test:serve:production\\"
            }
          }
        }
      }
    }
  },
  \\"defaultProject\\": \\"schematics-test\\"
}"
`;

exports[`Spartacus User schematics: ng-add Profile feature eager loading should import appropriate modules 1`] = `
"import { NgModule } from '@angular/core';
import { I18nConfig, provideConfig } from \\"@spartacus/core\\";
import { UserAccountModule } from \\"@spartacus/user/account\\";
import { userAccountTranslationChunksConfig, userAccountTranslations } from \\"@spartacus/user/account/assets\\";
import { UserAccountRootModule } from \\"@spartacus/user/account/root\\";
import { UserProfileModule } from \\"@spartacus/user/profile\\";
import { userProfileTranslationChunksConfig, userProfileTranslations } from \\"@spartacus/user/profile/assets\\";
import { UserProfileRootModule } from \\"@spartacus/user/profile/root\\";

@NgModule({
  declarations: [],
  imports: [
    UserAccountRootModule,
    UserAccountModule,
    UserProfileRootModule,
    UserProfileModule
  ],
  providers: [provideConfig(<I18nConfig>{
    i18n: {
      resources: userAccountTranslations,
      chunks: userAccountTranslationChunksConfig,
    },
  }),
  provideConfig(<I18nConfig>{
    i18n: {
      resources: userProfileTranslations,
      chunks: userProfileTranslationChunksConfig,
    },
  })
  ]
})
export class UserFeatureModule { }
"
`;

exports[`Spartacus User schematics: ng-add Profile feature general setup should add the feature using the lazy loading syntax 1`] = `
"import { NgModule } from '@angular/core';
import { CmsConfig, I18nConfig, provideConfig } from \\"@spartacus/core\\";
import { userAccountTranslationChunksConfig, userAccountTranslations } from \\"@spartacus/user/account/assets\\";
import { UserAccountRootModule, USER_ACCOUNT_FEATURE } from \\"@spartacus/user/account/root\\";
import { userProfileTranslationChunksConfig, userProfileTranslations } from \\"@spartacus/user/profile/assets\\";
import { UserProfileRootModule, USER_PROFILE_FEATURE } from \\"@spartacus/user/profile/root\\";

@NgModule({
  declarations: [],
  imports: [
    UserAccountRootModule,
    UserProfileRootModule
  ],
  providers: [provideConfig(<CmsConfig>{
    featureModules: {
      [USER_ACCOUNT_FEATURE]: {
        module: () =>
          import('@spartacus/user/account').then((m) => m.UserAccountModule),
      },
    }
  }),
  provideConfig(<I18nConfig>{
    i18n: {
      resources: userAccountTranslations,
      chunks: userAccountTranslationChunksConfig,
    },
  }),
  provideConfig(<CmsConfig>{
    featureModules: {
      [USER_PROFILE_FEATURE]: {
        module: () =>
          import('@spartacus/user/profile').then((m) => m.UserProfileModule),
      },
    }
  }),
  provideConfig(<I18nConfig>{
    i18n: {
      resources: userProfileTranslations,
      chunks: userProfileTranslationChunksConfig,
    },
  })
  ]
})
export class UserFeatureModule { }
"
`;

exports[`Spartacus User schematics: ng-add Profile feature general setup should install necessary Spartacus libraries 1`] = `
"{
  \\"name\\": \\"workspace\\",
  \\"version\\": \\"0.0.0\\",
  \\"scripts\\": {
    \\"ng\\": \\"ng\\",
    \\"start\\": \\"ng serve\\",
    \\"build\\": \\"ng build\\",
    \\"test\\": \\"ng test\\",
    \\"lint\\": \\"ng lint\\",
    \\"e2e\\": \\"ng e2e\\"
  },
  \\"private\\": true,
  \\"dependencies\\": {
    \\"@angular/animations\\": \\"~10.2.4\\",
    \\"@angular/common\\": \\"~10.2.4\\",
    \\"@angular/compiler\\": \\"~10.2.4\\",
    \\"@angular/core\\": \\"~10.2.4\\",
    \\"@angular/forms\\": \\"~10.2.4\\",
    \\"@angular/platform-browser\\": \\"~10.2.4\\",
    \\"@angular/platform-browser-dynamic\\": \\"~10.2.4\\",
    \\"@angular/router\\": \\"~10.2.4\\",
    \\"@angular/service-worker\\": \\"^10.1.0\\",
    \\"@ng-bootstrap/ng-bootstrap\\": \\"^7.0.0\\",
    \\"@ng-select/ng-select\\": \\"^5.0.9\\",
    \\"@ngrx/effects\\": \\"^10.0.0\\",
    \\"@ngrx/router-store\\": \\"^10.0.0\\",
    \\"@ngrx/store\\": \\"^10.0.0\\",
    \\"@spartacus/assets\\": \\"^3.2.0-next.2\\",
    \\"@spartacus/core\\": \\"^3.2.0-next.2\\",
    \\"@spartacus/storefront\\": \\"^3.2.0-next.2\\",
    \\"@spartacus/styles\\": \\"^3.2.0-next.2\\",
    \\"@spartacus/user\\": \\"^3.2.0-next.2\\",
    \\"angular-oauth2-oidc\\": \\"^10.0.1\\",
    \\"bootstrap\\": \\"^4.0\\",
    \\"i18next\\": \\"^19.3.4\\",
    \\"i18next-xhr-backend\\": \\"^3.2.2\\",
    \\"ngx-infinite-scroll\\": \\"^8.0.0\\",
    \\"rxjs\\": \\"~6.6.0\\",
    \\"tslib\\": \\"^2.0.0\\",
    \\"zone.js\\": \\"~0.10.2\\"
  },
  \\"devDependencies\\": {
    \\"@angular-devkit/build-angular\\": \\"~0.1002.1\\",
    \\"@angular-devkit/schematics\\": \\"^10.2.1\\",
    \\"@angular/cli\\": \\"~0.5.0\\",
    \\"@angular/compiler-cli\\": \\"~10.2.4\\",
    \\"@types/node\\": \\"^12.11.1\\",
    \\"@types/jasmine\\": \\"~3.5.0\\",
    \\"@types/jasminewd2\\": \\"~2.0.3\\",
    \\"codelyzer\\": \\"^6.0.0\\",
    \\"jasmine-core\\": \\"~3.6.0\\",
    \\"jasmine-spec-reporter\\": \\"~5.0.0\\",
    \\"karma\\": \\"~5.0.0\\",
    \\"karma-chrome-launcher\\": \\"~3.1.0\\",
    \\"karma-coverage-istanbul-reporter\\": \\"~3.0.2\\",
    \\"karma-jasmine\\": \\"~4.0.0\\",
    \\"karma-jasmine-html-reporter\\": \\"^1.5.0\\",
    \\"protractor\\": \\"~7.0.0\\",
    \\"ts-node\\": \\"~8.3.0\\",
    \\"tslint\\": \\"~6.1.0\\",
    \\"typescript\\": \\"~4.0.2\\"
  }
}
"
`;

exports[`Spartacus User schematics: ng-add Profile feature general setup styling should create a proper scss file 1`] = `"@import \\"@spartacus/user\\";"`;

exports[`Spartacus User schematics: ng-add Profile feature general setup styling should update angular.json 1`] = `
"{
  \\"$schema\\": \\"./node_modules/@angular/cli/lib/config/schema.json\\",
  \\"version\\": 1,
  \\"newProjectRoot\\": \\"\\",
  \\"projects\\": {
    \\"schematics-test\\": {
      \\"projectType\\": \\"application\\",
      \\"schematics\\": {
        \\"@schematics/angular:component\\": {
          \\"style\\": \\"scss\\"
        }
      },
      \\"root\\": \\"\\",
      \\"sourceRoot\\": \\"src\\",
      \\"prefix\\": \\"app\\",
      \\"architect\\": {
        \\"build\\": {
          \\"builder\\": \\"@angular-devkit/build-angular:browser\\",
          \\"options\\": {
            \\"outputPath\\": \\"dist/schematics-test\\",
            \\"index\\": \\"src/index.html\\",
            \\"main\\": \\"src/main.ts\\",
            \\"polyfills\\": \\"src/polyfills.ts\\",
            \\"tsConfig\\": \\"tsconfig.app.json\\",
            \\"aot\\": true,
            \\"assets\\": [
              \\"src/favicon.ico\\",
              \\"src/assets\\"
            ],
            \\"styles\\": [
              \\"src/styles.scss\\",
              \\"src/styles/spartacus/user.scss\\"
            ],
            \\"scripts\\": []
          },
          \\"configurations\\": {
            \\"production\\": {
              \\"fileReplacements\\": [
                {
                  \\"replace\\": \\"src/environments/environment.ts\\",
                  \\"with\\": \\"src/environments/environment.prod.ts\\"
                }
              ],
              \\"optimization\\": true,
              \\"outputHashing\\": \\"all\\",
              \\"sourceMap\\": false,
              \\"extractCss\\": true,
              \\"namedChunks\\": false,
              \\"extractLicenses\\": true,
              \\"vendorChunk\\": false,
              \\"buildOptimizer\\": true,
              \\"budgets\\": [
                {
                  \\"type\\": \\"initial\\",
                  \\"maximumWarning\\": \\"2mb\\",
                  \\"maximumError\\": \\"5mb\\"
                },
                {
                  \\"type\\": \\"anyComponentStyle\\",
                  \\"maximumWarning\\": \\"6kb\\",
                  \\"maximumError\\": \\"10kb\\"
                }
              ]
            }
          }
        },
        \\"serve\\": {
          \\"builder\\": \\"@angular-devkit/build-angular:dev-server\\",
          \\"options\\": {
            \\"browserTarget\\": \\"schematics-test:build\\"
          },
          \\"configurations\\": {
            \\"production\\": {
              \\"browserTarget\\": \\"schematics-test:build:production\\"
            }
          }
        },
        \\"extract-i18n\\": {
          \\"builder\\": \\"@angular-devkit/build-angular:extract-i18n\\",
          \\"options\\": {
            \\"browserTarget\\": \\"schematics-test:build\\"
          }
        },
        \\"test\\": {
          \\"builder\\": \\"@angular-devkit/build-angular:karma\\",
          \\"options\\": {
            \\"main\\": \\"src/test.ts\\",
            \\"polyfills\\": \\"src/polyfills.ts\\",
            \\"tsConfig\\": \\"tsconfig.spec.json\\",
            \\"karmaConfig\\": \\"karma.conf.js\\",
            \\"assets\\": [
              \\"src/favicon.ico\\",
              \\"src/assets\\"
            ],
            \\"styles\\": [
              \\"src/styles.scss\\",
              \\"src/styles/spartacus/user.scss\\"
            ],
            \\"scripts\\": []
          }
        },
        \\"lint\\": {
          \\"builder\\": \\"@angular-devkit/build-angular:tslint\\",
          \\"options\\": {
            \\"tsConfig\\": [
              \\"tsconfig.app.json\\",
              \\"tsconfig.spec.json\\",
              \\"e2e/tsconfig.json\\"
            ],
            \\"exclude\\": [
              \\"**/node_modules/**\\"
            ]
          }
        },
        \\"e2e\\": {
          \\"builder\\": \\"@angular-devkit/build-angular:protractor\\",
          \\"options\\": {
            \\"protractorConfig\\": \\"e2e/protractor.conf.js\\",
            \\"devServerTarget\\": \\"schematics-test:serve\\"
          },
          \\"configurations\\": {
            \\"production\\": {
              \\"devServerTarget\\": \\"schematics-test:serve:production\\"
            }
          }
        }
      }
    }
  },
  \\"defaultProject\\": \\"schematics-test\\"
}"
`;

exports[`Spartacus User schematics: ng-add Without features should not add the feature to the feature module 1`] = `
"import { NgModule } from '@angular/core';
import { AnonymousConsentsModule, AuthModule, CartModule, CartOccModule, CheckoutModule, CheckoutOccModule, CostCenterOccModule, ProductModule, ProductOccModule, UserOccTransitionalModule, UserTransitionalModule } from \\"@spartacus/core\\";
import { AddressBookModule, AnonymousConsentManagementBannerModule, AnonymousConsentsDialogModule, BannerCarouselModule, BannerModule, BreadcrumbModule, CartComponentModule, CartPageEventModule, CategoryNavigationModule, CheckoutComponentModule, CheckoutLoginModule, CmsParagraphModule, ConsentManagementModule, FooterNavigationModule, HamburgerMenuModule, HomePageEventModule, LinkModule, LoginRouteModule, LogoutModule, MyCouponsModule, MyInterestsModule, NavigationEventModule, NavigationModule, NotificationPreferenceModule, OrderCancellationModule, OrderConfirmationModule, OrderDetailsModule, OrderHistoryModule, OrderReturnModule, PaymentMethodsModule, ProductCarouselModule, ProductDetailsPageModule, ProductFacetNavigationModule, ProductImagesModule, ProductIntroModule, ProductListingPageModule, ProductListModule, ProductPageEventModule, ProductReferencesModule, ProductSummaryModule, ProductTabsModule, ReplenishmentOrderConfirmationModule, ReplenishmentOrderDetailsModule, ReplenishmentOrderHistoryModule, ReturnRequestDetailModule, ReturnRequestListModule, SearchBoxModule, SiteContextSelectorModule, StockNotificationModule, TabParagraphContainerModule, WishListModule } from \\"@spartacus/storefront\\";

@NgModule({
  declarations: [],
  imports: [
    // Auth Core
    AuthModule.forRoot(),
    LogoutModule,
    LoginRouteModule,
    // Basic Cms Components
    HamburgerMenuModule,
    SiteContextSelectorModule,
    LinkModule,
    BannerModule,
    CmsParagraphModule,
    TabParagraphContainerModule,
    BannerCarouselModule,
    CategoryNavigationModule,
    NavigationModule,
    FooterNavigationModule,
    BreadcrumbModule,
    // User Core,
    UserTransitionalModule,
    UserOccTransitionalModule,
    // User UI,
    AddressBookModule,
    PaymentMethodsModule,
    NotificationPreferenceModule,
    MyInterestsModule,
    StockNotificationModule,
    ConsentManagementModule,
    MyCouponsModule,
    // Anonymous Consents Core,
    AnonymousConsentsModule.forRoot(),
    // Anonymous Consents UI,
    AnonymousConsentsDialogModule,
    AnonymousConsentManagementBannerModule,
    // Product Core,
    ProductModule.forRoot(),
    ProductOccModule,
    // Product UI,
    ProductDetailsPageModule,
    ProductListingPageModule,
    ProductListModule,
    SearchBoxModule,
    ProductFacetNavigationModule,
    ProductTabsModule,
    ProductCarouselModule,
    ProductReferencesModule,
    ProductImagesModule,
    ProductSummaryModule,
    ProductIntroModule,
    // Cart Core,
    CartModule.forRoot(),
    CartOccModule,
    // Cart UI,
    CartComponentModule,
    WishListModule,
    // Checkout Core,
    CheckoutModule.forRoot(),
    CheckoutOccModule,
    CostCenterOccModule,
    // Checkout UI,
    CheckoutLoginModule,
    CheckoutComponentModule,
    OrderConfirmationModule,
    // Order,
    OrderHistoryModule,
    OrderDetailsModule,
    OrderCancellationModule,
    OrderReturnModule,
    ReturnRequestListModule,
    ReturnRequestDetailModule,
    ReplenishmentOrderHistoryModule,
    ReplenishmentOrderDetailsModule,
    ReplenishmentOrderConfirmationModule,
    // Page Events,
    NavigationEventModule,
    HomePageEventModule,
    CartPageEventModule,
    ProductPageEventModule,
  ]
})
export class SpartacusFeaturesModule { }
"
`;
